### PR TITLE
quay operator on ocs

### PR DESCRIPTION
**Pull request by parikshithb**
Verifies https://bugzilla.redhat.com/show_bug.cgi?id=1947796

Signed-off-by: Parikshith Byregowda <pbyregow@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/quay_operator.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        Deploys Quay operator
+
+        """
+        quay_operator_data = templating.load_yaml(file=constants.QUAY_SUB)
+        self.quay_operator_csv = quay_operator_data["spec"]["startingCSV"]
+        self.quay_operator = OCS(**quay_operator_data)
+        logger.info(f"Installing Quay operator: {self.quay_operator.name}")
+        self.quay_operator.create()
+        for quay_pod in TimeoutSampler(
+            300, 10, get_pod_name_by_pattern, "quay-operator", self.namespace
```
Comment (PrasadDesala) [inline ocs_ci/ocs/quay_operator.py:52]: Please move it to constants

File: ocs_ci/ocs/quay_operator.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        ):
+            if quay_pod:
+                assert self.quay_pod_obj.wait_for_resource(
```
Comment (PrasadDesala) [inline ocs_ci/ocs/quay_operator.py:55]: wait for resource will throws exception after the timeout, do we still need assert here?

File: ocs_ci/ocs/quay_operator.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        """
+        if not helpers.get_default_storage_class():
+            patch = ' \'{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}\' '
+            run_cmd(
+                f"oc patch storageclass {self.sc_name} "
+                f"-p {patch} "
+                f"--request-timeout=120s"
+            )
+            self.sc_default = True
```
Comment (PrasadDesala) [inline ocs_ci/ocs/quay_operator.py:75]: we already have an existing function to change the default storage class to a given SC.
change_default_storageclass() in helpers.py file, please use it here.

File: ocs_ci/ocs/quay_operator.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        Quay operator teardown
+
+        """
+        if self.sc_default:
+            patch = ' \'{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}\' '
+            run_cmd(
+                f"oc patch storageclass {self.sc_name} "
+                f"-p {patch} "
+                f"--request-timeout=120s"
+            )
```
Comment (PrasadDesala) [inline ocs_ci/ocs/quay_operator.py:127]: Same as above, we already have an existing function to change the default storage class to a given SC.
change_default_storageclass() in helpers.py file, please use it here.

File: ocs_ci/ocs/quay_operator.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.namespace = "openshift-operators"
+        self.quay_operator = None
+        self.quay_registry = None
+        self.quay_pod_obj = OCP(kind="pod", namespace=self.namespace)
```
Comment (prsurve) [inline ocs_ci/ocs/quay_operator.py:31]: make use of constant

File: ocs_ci/ocs/quay_operator.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    Quay operator class
+
+    """
+
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.namespace = "openshift-operators"
```
Comment (prsurve) [inline ocs_ci/ocs/quay_operator.py:28]: Add namespace to constant.py

File: ocs_ci/ocs/quay_operator.py (hunk 5 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py]: done

File: ocs_ci/ocs/quay_operator.py (hunk 6 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py]: done

File: ocs_ci/ocs/quay_operator.py (hunk 2 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py]: removed assert.

File: ocs_ci/ocs/quay_operator.py (hunk 1 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py]: done

File: ocs_ci/ocs/quay_operator.py (hunk 3 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py:75]: I am not trying to change the default storageclass. I am just checking if there is an existing default storageclass, if so use or patch a storageclass.

File: ocs_ci/ocs/quay_operator.py (hunk 4 above)
Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py:127]: same as above

**Review by Akarsha-rai — DISMISSED** [review]
LGTM

File: ocs_ci/ocs/quay_operator.py (hunk 3 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/quay_operator.py:75]: ok, can we use default_storage_class here?

Comment (parikshithb) [reply ocs_ci/ocs/quay_operator.py:75]: Don't think so. Default_storage_class gets sc object based on the interface type given which is not required here.

**Review by Akarsha-rai — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**